filebeat/input/journald/input.go

```diff
@@ -272,7 +235,26 @@ func (r *readerAdapter) Next() (reader.Message, error) {
 
 	created := time.Now()
 
-	content := []byte(data.Fields["MESSAGE"])
+	// Journald documents that 'MESSAGE' is always a string,
+	// see https://www.man7.org/linux/man-pages/man7/systemd.journal-fields.7.html.
+	// However while testing 'journalctl -o json' outputs the 'MESSAGE'
+	// like [1, 2, 3, 4]. Which seems to be the result of a binary encoding
+	// of a journal field (see https://systemd.io/JOURNAL_NATIVE_PROTOCOL/).
+	//
+	// Trying to be smart and convert the contents into string
+	// byte by byte did not work well because one test case contained
+	// control characters and new line characters.
+	// To avoid issues later in the ingestion pipeline we just convert
+	// the whole thing to a string using fmt.Sprint.
+	//
+	// Look at 'pkg/journalctl/testdata/corner-cases.json'
+	// for some real world examples.
```

Comment:
I added a comment (https://github.com/elastic/beats/pull/40061#discussion_r1664887758) on this test data explaining what happens.

After trying to convert it to string and make sense of the result, I just gave up and decided to use `fmt.Sprint` to avoid any encoding/control character issue.

The fact that `journalctl` does not try to print it as string even though it documents `MESSAGE` is a human readable string, makes me believe we also should not try to automagically convert it into a string.